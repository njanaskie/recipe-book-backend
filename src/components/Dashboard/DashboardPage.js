import React, { useReducer } from 'react'
import dishesReducer from '../../reducers/dishes'
import DishesContext from '../../../context/dishes-context'
import PantryContext from '../../../context/pantry-context'
import FiltersContext from '../../../context/filters-context'
import pantryReducer from '../../reducers/pantry'
import filtersReducer from '../../reducers/filters'
import DashboardList from '../Dashboard/DashboardList'


const DashboardPage = () => {
    const [dishes, dishDispatch] = useReducer(dishesReducer, [])
    const [pantryIngredients, pantryDispatch] = useReducer(pantryReducer, [])
    const [filters, filtersDispatch] = useReducer(filtersReducer)

    return (
        <FiltersContext.Provider value={{ filters, filtersDispatch }}>
            <DishesContext.Provider value={{ dishes, dishDispatch }}>
                <PantryContext.Provider value={{ pantryIngredients, pantryDispatch }}>
                    <DashboardList />
                </PantryContext.Provider>
            </DishesContext.Provider>
        </FiltersContext.Provider>
    )

};

export default DashboardPage;