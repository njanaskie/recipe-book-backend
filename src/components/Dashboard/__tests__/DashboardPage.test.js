import React from 'react'
import { shallow } from 'enzyme'
import DashboardPage from '../../Dashboard/DashboardPage'

test('should render DashboardPage correctly',() => {
    const wrapper = shallow(<DashboardPage />)
    expect(wrapper).toMatchSnapshot()
})