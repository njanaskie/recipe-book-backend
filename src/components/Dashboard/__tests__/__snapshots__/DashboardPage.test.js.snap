// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardPage correctly 1`] = `
<[object Object]
  value={
    Object {
      "filters": undefined,
      "filtersDispatch": [Function],
    }
  }
>
  <[object Object]
    value={
      Object {
        "dishDispatch": [Function],
        "dishes": Array [],
      }
    }
  >
    <[object Object]
      value={
        Object {
          "pantryDispatch": [Function],
          "pantryIngredients": Array [],
        }
      }
    >
      <h3>
        Dashboard Page
      </h3>
      <DashboardList />
    </[object Object]>
  </[object Object]>
</[object Object]>
`;
